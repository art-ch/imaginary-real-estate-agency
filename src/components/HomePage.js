import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';

import { getRealEstate } from '../redux/ducks/getRealEstate';

import HomePageWrapper from '../styled/HomePage';
import { Card, Container, Row, Col, Button, Spinner } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

const HomePage = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getRealEstate());
  }, []);

  const realEstateList = useSelector(
    (state) => state.getRealEstateReducer.realEstate
  );

  return (
    <HomePageWrapper>
      <h1 className="page-title">Imaginary Real Estate Agency</h1>
      {/* {loading && <Spinner animation="border" variant="primary" />} */}
      <Container>
        <Row>
          {realEstateList.map((house) => {
            const { id, title, image, price, address } = house;
            return (
              <Col key={id}>
                <Card className="mx-auto mb-4 fluid">
                  <Card.Img src={image} className="card-image" />
                  <Card.Body>
                    <Card.Title>{title}</Card.Title>
                    <Card.Text>{address}</Card.Text>
                    <Card.Title className="price-tag">{price}</Card.Title>
                    <Link to={`/houses/house/${id}`}>Details</Link>
                  </Card.Body>
                </Card>
              </Col>
            );
          })}
        </Row>
      </Container>
    </HomePageWrapper>
  );
};

export default HomePage;
