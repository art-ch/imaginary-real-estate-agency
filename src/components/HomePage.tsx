import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Card, Container, Row, Col, Button, Spinner } from 'react-bootstrap';

import HomePageWrapper from '../styled/HomePage';

import { getRealEstate } from '../redux/ducks/getRealEstate';
import 'bootstrap/dist/css/bootstrap.min.css';
import { HouseSchema } from '../interfaces';

const HomePage = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(getRealEstate());
    // eslint-disable-next-line
  }, []);

  const realEstateList = useSelector(
    (state: any) => state.getRealEstateReducer.realEstate
  );

  return (
    <HomePageWrapper>
      <h1 className="page-title">Imaginary Real Estate Agency</h1>
      {realEstateList.length === 0 && (
        <Spinner animation="border" variant="primary" />
      )}
      <Container>
        <Row className="g-lg-3 g-xl-4 g-5">
          {realEstateList.map((house: HouseSchema) => {
            const { id, title, image, price, address } = house;
            return (
              <Col className="col-lg-4 col-xl-3" key={id}>
                <Card className="mx-auto">
                  <a href={image}>
                    <Card.Img src={image} className="card-image" />
                  </a>
                  <Card.Body>
                    <Card.Title>{title}</Card.Title>
                    <Card.Text>{address}</Card.Text>
                    <Card.Title className="price-tag">{price}</Card.Title>
                    <Button onClick={() => navigate(`/houses/house/${id}`)}>
                      Details
                    </Button>
                  </Card.Body>
                </Card>
              </Col>
            );
          })}
        </Row>
      </Container>
    </HomePageWrapper>
  );
};

export default HomePage;
