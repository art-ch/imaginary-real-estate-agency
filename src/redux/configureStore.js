import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import getRealEstateReducer from './ducks/getRealEstate';
import getHouseReducer from './ducks/getHouse';
import { watcherSaga } from './sagas/rootSaga';

const reducer = combineReducers({ getRealEstateReducer, getHouseReducer });

const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

const store = createStore(reducer, {}, applyMiddleware(...middleware));

sagaMiddleware.run(watcherSaga);

export default store;
