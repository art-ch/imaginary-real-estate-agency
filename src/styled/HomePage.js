import styled from 'styled-components';

const HomePageWrapper = styled.main`
  text-align: center;
  .card {
    width: 300px;
    .card-image {
      height: 15rem;
    }
    .price-tag {
      margin: 1rem 0;
      font-size: 32px;
    }
  }
`;

export default HomePageWrapper;
