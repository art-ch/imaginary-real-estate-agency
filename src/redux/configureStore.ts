import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import getRealEstateReducer from './ducks/getRealEstate';
import getHouseReducer from './ducks/getHouse';
import { watcherSaga } from './rootSaga';

const reducer = combineReducers({ getRealEstateReducer, getHouseReducer });

const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

const store = createStore(reducer, {}, applyMiddleware(...middleware));

sagaMiddleware.run(watcherSaga);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
