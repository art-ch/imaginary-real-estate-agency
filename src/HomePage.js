import React, { useEffect, useContext } from 'react';
import styled from 'styled-components';
import { Context } from './Context';
import { Card, Container, Row, Col, Button, Spinner } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';

const HomePage = () => {
  const { fetchRealEstate, handlePrice, loading, realEstateList } =
    useContext(Context);

  useEffect(() => {
    fetchRealEstate();
    // eslint-disable-next-line
  }, []);

  return (
    <Wrapper>
      <h1 className="page-title">Imaginary Real Estate Agency</h1>
      {loading && <Spinner animation="border" variant="primary" />}
      <Container>
        <Row>
          {realEstateList.map((house) => {
            const { id, title, image, price, address } = house;
            return (
              <Col>
                <Card
                  style={{ width: '300px' }}
                  className="mx-auto mb-2 md-fluid"
                  key={id}
                >
                  <Card.Img src={image} className="card-image" />
                  <Card.Body>
                    <Card.Title>{title}</Card.Title>
                    <Card.Text>{address}</Card.Text>
                    <Card.Title className="price-tag">
                      {handlePrice(price)}
                    </Card.Title>
                    <Button>Details</Button>
                  </Card.Body>
                </Card>
              </Col>
            );
          })}
        </Row>
      </Container>
    </Wrapper>
  );
};

export default HomePage;

const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  .container {
    margin: 0 auto;
  }
  .card-image {
    height: 15rem;
    object-fit: cover;
  }
  .price-tag {
    margin: 1rem 0;
    font-size: 32px;
  }

  @media (max-width: 767px) {
    .small-screen-centering {
      width: 300px;
    }
  }
`;
