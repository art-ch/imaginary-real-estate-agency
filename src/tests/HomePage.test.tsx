import { Provider } from 'react-redux';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import HomePage from '../components/HomePage';

const mockStore = configureStore([]);

describe('<HomePage />', () => {
  let store;
  let component: any;

  beforeEach(() => {
    store = mockStore({
      getRealEstateReducer: {
        realEstate: [
          {
            id: '1',
            title: 'House1',
            price: 228000,
            address: '1997 Yates Ave, Montebello, CA',
            image: 'image',
            description: 'desc',
            name: 'Ronnie Polla'
          }
        ]
      }
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Routes>
            <Route path="/" element={<HomePage />} />
          </Routes>
        </MemoryRouter>
      </Provider>
    );
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
