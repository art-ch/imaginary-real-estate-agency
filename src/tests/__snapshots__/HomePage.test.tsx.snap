// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match snapshot 1`] = `
<main
  className="sc-bdvvtL cjxUAN"
>
  <h1
    className="page-title"
  >
    Imaginary Real Estate Agency
  </h1>
  <div
    className="container"
  >
    <div
      className="g-lg-3 g-xl-4 g-5 row"
    >
      <div
        className="col-lg-4 col-xl-3 col"
      >
        <div
          className="mx-auto card"
        >
          <a
            href="image"
          >
            <img
              className="card-img card-image"
              src="image"
            />
          </a>
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              House1
            </div>
            <p
              className="card-text"
            >
              1997 Yates Ave, Montebello, CA
            </p>
            <div
              className="price-tag card-title h5"
            >
              228000
            </div>
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
